tests/functional/testOnboarding.js

```diff
@@ -277,4 +279,65 @@ describe('Onboarding', function() {
     assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.ONBOARDING_VIEW, 'currentIndex'), 3);
     assert.equal(await vpn.getSetting('onboardingStep'), 3);
   });
+
+  it.only('Verify data collection checkbox and setting are decoupled', async () => {
+
+    //This function completes onboarding, and checks that the gleanEnabled setting remains enabled throughout the flow
+    async function completeOnboarding() {
+      await vpn.waitForQueryAndClick(queries.screenOnboarding.DATA_NEXT_BUTTON.visible());
+      await vpn.waitForQuery(queries.screenOnboarding.STEP_NAV_STACK_VIEW.ready());
+      assert.equal(await vpn.getSetting('gleanEnabled'), true);
+      await vpn.waitForQueryAndClick(queries.screenOnboarding.PRIVACY_NEXT_BUTTON.visible());
+      await vpn.waitForQuery(queries.screenOnboarding.STEP_NAV_STACK_VIEW.ready());
+      assert.equal(await vpn.getSetting('gleanEnabled'), true);
+      await vpn.waitForQueryAndClick(queries.screenOnboarding.DEVICES_NEXT_BUTTON.visible());
+      await vpn.waitForQuery(queries.screenOnboarding.STEP_NAV_STACK_VIEW.ready());
+      assert.equal(await vpn.getSetting('gleanEnabled'), true);
+      await vpn.waitForQueryAndClick(queries.screenOnboarding.START_NEXT_BUTTON.visible());
+      await vpn.waitForQuery(queries.screenHome.SCREEN.visible());
+      assert.equal(await vpn.getSetting('onboardingCompleted'), true);
+    }
+
+    //Ensure we are in onboarding
+    await vpn.waitForQuery(queries.screenOnboarding.DATA_SLIDE.visible());
+
+    //Ensure that the data collection checkbox and internal setting are decoupled
+    //We will finish onboarding with the data collection checkbox being unchecked
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.DATA_CHECKBOX, 'isChecked'), 'false');
+    assert.equal(await vpn.getSetting('gleanEnabled'), true);
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.DATA_CHECKBOX.visible());
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.DATA_CHECKBOX, 'isChecked'), 'true');
+    assert.equal(await vpn.getSetting('gleanEnabled'), true);
```

Comment:
Is this back and forth checkbox toggle necessary here for the sake of the test flow? Asking because we are testing checking this on line 30 in a different test.

Reply:
Note: This also should not be necessary by ending the test case here and starting new ones for the other checks.